File: TheRayCode/JavaScript/Creational/Factory/Example1/ConcreteCreator.js
// ConcreteCreator.js
// Purpose: Concrete creator class implementing factoryMethod

import Creator from './Creator.js';
import ConcreteProduct from './ConcreteProduct.js';

class ConcreteCreator extends Creator {
    factoryMethod() {
        return new ConcreteProduct();
    }
}

export default ConcreteCreator;

File: Python/Creational/Factory/Example1/ConcreteCreator.py
# Import the Creator participant.
from Creator import Creator

# Import the Product participant.
from Product import Product

# Import the ConcreteProduct participant.
from ConcreteProduct import ConcreteProduct


# Define the ConcreteCreator participant.
class ConcreteCreator(Creator):

    # Override FactoryMethod.
    def FactoryMethod(self) -> Product:

        # Return a ConcreteProduct instance.
        return ConcreteProduct()